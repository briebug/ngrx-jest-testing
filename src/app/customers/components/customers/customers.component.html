<ul>
  <li *ngFor="let customer of customers">Customer: {{customer.id}} - {{customer.name}}
  </li>
</ul>
