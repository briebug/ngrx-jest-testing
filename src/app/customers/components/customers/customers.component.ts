import { Component, OnInit, Input } from '@angular/core';
import { CustomerActions } from '@state/customer/customer.actions';
import { Customer } from '@state/customer/customer.model';

@Component({
  selector: 'app-customers',
  templateUrl: './customers.component.html',
  styleUrls: ['./customers.component.css']
})
export class CustomersComponent implements OnInit {
  @Input() customers: Customer[];

  constructor() { }

  ngOnInit() { }
}
