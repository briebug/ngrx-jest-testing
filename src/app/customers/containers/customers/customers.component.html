<app-customers [customers]="customers | async"></app-customers>
