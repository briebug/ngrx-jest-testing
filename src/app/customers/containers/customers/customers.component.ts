import { Component, OnInit } from '@angular/core';
import { Observable } from 'rxjs/Observable';
import { Store, select } from '@ngrx/store';

import { Customer } from '@state/customer/customer.model';
import { AppState } from '@state/app.interfaces';
import { Load } from '../../../state/customer/customer.actions';
import { getAllCustomers } from '../../../state/customer';

@Component({
  selector: 'app-customers-container',
  templateUrl: './customers.component.html',
  styleUrls: ['./customers.component.css']
})
export class CustomersComponentContainer implements OnInit {
  customers$: Observable<Customer[]>;

  constructor(private store: Store<AppState>) {
    this.store.dispatch(new Load());
  }

  ngOnInit() {
    this.customers$ = this.store.pipe(select(getAllCustomers));
  }
}
