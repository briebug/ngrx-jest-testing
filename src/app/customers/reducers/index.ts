import { createSelector, createFeatureSelector } from '@ngrx/store';

import * as fromCustomers from './customer';

export interface CustomersState {
  customers: fromCustomers.State;
}

export const reducers = {
  customers: fromCustomers.reducer
};

export const getCustomersState = createFeatureSelector<CustomersState>(
  'customers'
);

export const getCustomerEntitiesState = createSelector(
  getCustomersState,
  state => state.customers
);

export const getSelectedCustomerId = createSelector(
  getCustomerEntitiesState,
  fromCustomers.getSelectedId
);

export const {
  selectIds: getCustomerIds,
  selectEntities: getCustomerEntities,
  selectAll: getAllCustomers,
  selectTotal: getTotalCustomers
} = fromCustomers.adapter.getSelectors(getCustomerEntitiesState);

export const getSelectedCustomer = createSelector(
  getCustomerEntities,
  getSelectedCustomerId,
  (entities, selectedId) => {
    return selectedId && entities[selectedId];
  }
);
