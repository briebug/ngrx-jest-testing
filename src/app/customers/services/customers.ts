import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { map } from 'rxjs/operators';
import { Customer } from '../models/customer';

@Injectable()
export class CustomersService {
  private API_PATH = '/api/customers';

  constructor(private http: HttpClient) {}

  load(): Observable<Customer[]> {
    return this.http.get<Customer[]>(this.API_PATH);
  }
}
