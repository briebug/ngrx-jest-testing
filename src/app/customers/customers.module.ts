import { CommonModule } from '@angular/common';
import { ModuleWithProviders, NgModule, Optional, SkipSelf } from '@angular/core';

import { CustomersComponentContainer } from './containers/customers/customers.component';
import { CustomersComponent } from './components/customers/customers.component';

@NgModule({
  imports: [
    CommonModule
  ],
  declarations: [
    CustomersComponentContainer,
    CustomersComponent
  ]
})

export class CustomersModule {

  static forRoot(): ModuleWithProviders {
    return {
      ngModule: CustomersModule,
      providers: [
      ]
    };
  }

  constructor(@Optional() @SkipSelf() parentModule: CustomersModule) {
    if (parentModule) {
      throw new Error(
        'CustomersModule is already loaded. Import it in the AppModule only');
    }
  }
}
