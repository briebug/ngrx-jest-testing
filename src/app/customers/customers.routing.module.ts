import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { CustomersModule } from './customers.module';
import { CustomersComponentContainer } from './containers/customers/customers.component';

const routes: Routes = [
  {
    path: '',
    component: CustomersComponentContainer
  }
];

@NgModule({
  imports: [CustomersModule.forRoot(), RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class CustomersRoutingModule { }
