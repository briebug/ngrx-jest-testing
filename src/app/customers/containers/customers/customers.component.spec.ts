import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { of } from 'rxjs/observable/of';
import { configureTests, ConfigureFn } from '../../../../test-config.helper'

import { CustomersComponentContainer } from './customers.component';
import { CustomersComponent } from '../../components/customers/customers.component';

describe('CustomersComponentContainer', () => {
  let component: CustomersComponentContainer,
    fixture: ComponentFixture<CustomersComponentContainer>,
    customers = [
      { id: 1, name: 'test1' },
      { id: 2, name: 'test2' },
      { id: 3, name: 'test3' }
    ];

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [
            CustomersComponentContainer,
            CustomersComponent
          ]
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(CustomersComponentContainer);
        component = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('renders markup to snapshot', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
