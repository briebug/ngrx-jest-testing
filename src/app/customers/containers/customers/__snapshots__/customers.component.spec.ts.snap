// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersComponentContainer renders markup to snapshot 1`] = `
<app-customers-container
  customers$={[Function TestHotObservable]}
  store={[Function Object]}
>
  <h2>
    Customers
  </h2><app-customers>
    <ul>
      
    </ul>
  </app-customers>
</app-customers-container>
`;
